pallets/identity/src/lib.rs

```diff
@@ -261,14 +277,22 @@ decl_module! {
         // this is needed only if you are using events in your module
         fn deposit_event() = default;
 
+        fn on_runtime_upgrade() -> Weight {
+            if <StorageVersion>::get().is_none() {
+                // Upgrade to V1.
+                <CddAuthForPrimaryKeyRotation>::put(Self::cdd_auth_for_master_key_rotation());
```

Comment:
you can also free the storage here for the old data type as it has no usage anymore.